stage3/buf.C

```diff
@@ -68,7 +68,37 @@ const Status BufMgr::unPinPage(File *file, const int PageNo, const bool dirty) {
 
 }
 
-const Status BufMgr::allocPage(File *file, int &pageNo, Page *&page) {}
+const Status BufMgr::allocPage(File *file, int &pageNo, Page *&page) {
+  if (file == nullptr)
+    return UNIXERR;
+  auto status = file->allocatePage(pageNo);
+  if (status != OK) 
+    return status;
+  
+  int frameNo = -1;
+  //obtain a buffer pool frame
+  status = allocBuf(frameNo);
+  if (status != OK) 
+    return status;
+  
+  //an entry is inserted into the hash table
+  if (hashTable == nullptr)
+    return HASHTBLERROR;
+  status = hashTable->insert(file, pageNo, frameNo);
+  if (status != OK)
+    return status;
+  
+  //The BufDesc class is used to keep track of the state of each frame in the buffer pool.
+  //thus we have to update the correct frame
+  //in the buff table
+  if (bufTable[frameNo] == nullptr) {
+    return UNIXERR;
+  }
+  bufTable[frameNo]->Set(file, pageNo);
+  //iffy on this line
+  page = bufPool[pageNo];
```

Comment:
`bufPool` is an array of `Page`s, so the RHS is a `Page` object instead of address of a `Page`.  
Should be `page = bufPool + pageNo` or `page = &bufPool[pageNo]`